src/goto-programs/goto_convert.cpp

```diff
@@ -344,14 +344,40 @@ void goto_convertt::convert_label(
 
   goto_programt tmp;
 
-  // magic thread creation label?
+  // magic thread creation label.
+  // The "__CPROVER_ASYNC_" signals the start
+  // of a sequence of instructions
+  // that can be executed in parallel, i.e a
+  // new thread.
+  //
+  // note: this technique for using a label as a way
+  // to instrument a thread block is only viable
+  // for C programs (see ansi-c/library/pthread_lib/pthread.c)
```

Comment:
No. `codet` is independent of C and `code_labelt` could be generated in any front-end.